### PR TITLE
feat(sca): add is public fix version to sbom report 

### PR DESCRIPTION
**Pull request by ayajbara**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/output/csv.py [hunk 1]
```diff
@@ -82,6 +84,10 @@ def add_sca_package_resources(self, resource: Record | ExtraResource, git_org: s
             CheckType.SCA_IMAGE: self.container_rows
         }
 
+        is_private_fix = resource.vulnerability_details.get("is_private_fix")
+        public_fix_version_suffix = " (Public)" if is_private_fix is False else ""
+        lowest_fix_version = get_lowest_fix_version(resource.vulnerability_details)
+        fix_version = lowest_fix_version + public_fix_version_suffix if lowest_fix_version and lowest_fix_version != UNFIXABLE_VERSION else lowest_fix_version
```
Comment (gruebel) [inline checkov/common/output/csv.py:90]: can you move it to a tiny method

File: checkov/common/output/cyclonedx.py [hunk 2]
```diff
@@ -380,6 +381,10 @@ def create_cve_vulnerability(self, resource: Record, component: Component) -> Vu
             method = VulnerabilityScoreSource.get_from_vector(vector)
             vector = method.get_localised_vector(vector)
 
+        is_private_fix = resource.vulnerability_details.get("is_private_fix")
+        public_fix_version_prefix = "No private fix available. " if is_private_fix is False else ""
+        status = resource.vulnerability_details.get("status")
+        recommendation = public_fix_version_prefix + status if status and status != UNFIXABLE_VERSION else status
```
Comment (gruebel) [inline checkov/common/output/cyclonedx.py:387]: same here

File: checkov/common/sca/commons.py [hunk 3]
```diff
@@ -47,3 +48,24 @@ def normalize_twistcli_language(language: str) -> str:
...
+
+
+def get_lowest_fix_version(vulnerability_details: dict[str, Any]) -> str:
+    if "cveStatus" in vulnerability_details:
+        return str(vulnerability_details["cveStatus"])
+
+    lowest_fixed_version = UNFIXABLE_VERSION
+    package_version = vulnerability_details["package_version"]
+    fixed_versions: list[packaging_version.Version | packaging_version.LegacyVersion] = []
+    status = vulnerability_details.get("status") or "open"
```
Comment (gruebel) [inline checkov/common/sca/commons.py:60]: can you create a global const for status `open` next to `UNFIXABLE_VERSION` at the top

**Review by gruebel — APPROVED** [review]
nice work 🥇

**Review by AdamDev — APPROVED** [review]

**Pull request merged**